docs/application/dotnet/guides/nui/widget.md

```diff
@@ -0,0 +1,255 @@
+|--------------------|-----------------------------------------------------------------------------------------------|
+| `OnCreate()`       | This function is called after the widget instance is created.                                 |
+| `OnTerminate()`    | This function is called after the widget instance is terminated.                              |
+| `OnPause()`        | This function is called when the widget is invisible.                                         |
+| `OnResume()`       | This function is called when the widget is visible.                                           |
+| `OnResize()`       | This function is called before the widget size is changed.                                    |
+| `OnUpdate()`       | This function is called when an event for updating the widget is received.                    |
+| `SetContentInfo()` | This function sends data about current state of widget instance to a widget viewer application. |
+
+Most of functions are used for managing widget instance life-cycle. You can describe what will be done on each lifecycle functions.
+
+Widget instance can send data to widget viewer application using `SetContentInfo()` function. If you want to save the current state of widget before deleting it, you can transmit the current status to the viewer using `SetContentInfo()`.
+
+## Create a NUI Widget Application
+
+If you want to use NUI for your widget application, you must create the application with the [NUI Widget Application](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.NUIWidgetApplication.html) class. 
+
+To create a NUI widget application:
+1. To use the methods and properties of the [Tizen.NUI](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.html) namespace, include the following code:
+
+   ```csharp
+   using Tizen.NUI;
+   ```
+
+2. Define your widget application class, which is inherited from the [NUIWidgetApplication](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.NUIWidgetApplication.html) class:
+
+   ```csharp
+   class Program : NUIWidgetApplication
+   ```
+3. The widget application starts with the `Main()` function, which creates and initializes the application. The `Run()` method of the [NUIWidgetApplication](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.NUIWidgetApplication.html) class is used to start the application event-loop.
+
+   The `NUIWidgetApplication` class provides two kinds of constructors:
+   - For using the `NUIWidgetApplication(Type type)` constructor, in case that widget application's ID is the same as the application ID.
+      ```csharp
+        static void Main(string[] args)
+        {
+            var app = new program(typeof(MyWidget));
+            app.Run(args);
+        }    
+      ```
+
+   - For using the `NUIWidgetApplication(Dictionary< Type, string > typeInfo) ` constructor, in case that your widget applications has multiple widget classes.
```

Comment:
```suggestion
 - For using the `NUIWidgetApplication(Dictionary< Type, string > typeInfo) ` constructor, in case your widget applications have multiple widget classes.
```